### PR TITLE
Improve s3 backup store client reliability

### LINKED ISSUE #10547: `IN_PROGRESS` backup is stuck
**Describe the bug**

After starting a backup, the backup is seemingly stuck `IN_PROGRESS`. There were no node restarts or leader changes during this period.

**To Reproduce**

Apparently just a regular `POST /actuator/backup/{id}`

**Expected behavior**

Backup should succeed or fail eventually.

[Thread dump](https://github.com/camunda/zeebe/files/9666674/threads.txt)
[Heap dump](https://drive.google.com/file/d/1AAWRvX6pKincF0MupsWTZgUPmbtRqxeF/view?usp=sharing)

### PR DESCRIPTION
**Pull request by lenaschoenburg**
This adds the ability to set an overall api call timeout which appears to be the only thing that fixes #10547. Additionally, some other configuration for the client is updated to smooth out load spikes when taking backups and to automatically adjust various other parameters based on the detected environment.

### CODE REVIEW

File: broker/src/main/java/io/camunda/zeebe/broker/system/configuration/backup/S3BackupStoreConfig.java [hunk 1]
```diff
@@ -17,6 +18,7 @@ public class S3BackupStoreConfig implements ConfigurationEntry {
   private String region;
   private String accessKey;
   private String secretKey;
+  private Duration apiCallTimeout;
```
Comment (deepthidevaki) [inline broker/src/main/java/io/camunda/zeebe/broker/system/configuration/backup/S3BackupStoreConfig.java:21]: :wrench: I suggest to provide a default value.

**Review by deepthidevaki — CHANGES_REQUESTED** [review]
Thanks. Looks good to me.

:x: Please document the new parameter in config templates in `dist/config`

File: broker/src/main/java/io/camunda/zeebe/broker/system/configuration/backup/S3BackupStoreConfig.java (hunk 1 above)
Comment (oleschoenburg) [reply broker/src/main/java/io/camunda/zeebe/broker/system/configuration/backup/S3BackupStoreConfig.java]: I've set it to 180 seconds. If we assume that the largest file of a backup is a 128MiB segment this requires less than 1MiB/s throughput which should be reasonable.

**Review by deepthidevaki — APPROVED** [review]
:tada:

**Comment by oleschoenburg** [comment]
bors retry

**Comment by oleschoenburg** [comment]
bors retry

**Comment by oleschoenburg** [comment]
bors retry

**Pull request merged**